spec/manifolds/cli_spec.rb

```diff
@@ -3,86 +3,102 @@
       end
 
-      it "creates a tables directory within the project" do
+      it "creates the expected project structure" do
         expect(Dir.exist?("./projects/#{sub_project_name}/tables")).to be true
-      end
-
-      it "creates a routines directory within the project" do
         expect(Dir.exist?("./projects/#{sub_project_name}/routines")).to be true
       end
 
-      it "creates a manifold.yml file" do
-        expect(File.exist?("./projects/#{sub_project_name}/manifold.yml")).to be true
-      end
-
-      it "writes the manifold.yml file with dimensions" do
-        expect(File.read("./projects/#{sub_project_name}/manifold.yml")).to include("dimensions")
-      end
-
-      it "writes the manifold.yml file with metrics" do
-        config = File.read("./projects/#{sub_project_name}/manifold.yml")
-        expect(config).to include("metrics")
+      it "creates a manifold configuration file" do
+        config_file = "./projects/#{sub_project_name}/manifold.yml"
+        expect(File.exist?(config_file)).to be true
+        expect(File.read(config_file)).to include("vectors:")
+        expect(File.read(config_file)).to include("metrics:")
       end
     end
 
     context "when outside an umbrella project" do
-      subject(:cli_with_stdout) { described_class.new(logger: Logger.new($stdout)) }
+      let(:cli_with_stdout) { described_class.new(logger: Logger.new($stdout)) }
 
-      it "does not allow adding projects and logs an error" do
-        expect do
-          cli_with_stdout.add("Pages")
-        end.to output(/Not inside a Manifolds umbrella project./).to_stdout
+      it "indicates that the command must be run within a project" do
+        FakeFS do
+          expect { cli_with_stdout.add(sub_project_name) }
+            .to output(/Not inside a Manifolds umbrella project/).to_stdout
+        end
       end
     end
   end
 
-  describe "#generate" do
-    subject(:cli) { described_class.new(logger: null_logger) }
+  describe "vectors" do
+    describe "add" do
+      subject(:cli) { vectors_command.new }
 
-    before do
-      allow(Manifolds::Services::BigQueryService).to receive(:new).and_return(bq_service)
-      allow(bq_service).to receive(:generate_dimensions_schema)
-    end
+      let(:vector_name) { "Page" }
+      let(:vectors_command) { described_class.new.class.subcommand_classes["vectors"] }
 
-    it "calls generate_dimensions_schema on bq service with correct project name" do
-      cli.generate("Pages", "bq")
-      expect(bq_service).to have_received(:generate_dimensions_schema).with("Pages")
+      context "when adding a vector within an umbrella project" do
+        before do
+          FileUtils.mkdir_p("./vectors")
+          cli.add(vector_name)
+        end
+
+        it "creates a vector configuration file" do
+          expect(File.exist?("./vectors/page.yml")).to be true
+          expect(File.read("./vectors/page.yml")).to include("attributes:")
+        end
```

Comment:
Actually parse the file instead of looking for substrings. Also, the "file exists" assertion is redundant.